src/modules/coin/mod.rs

```diff
@@ -1,32 +1,59 @@
+use actix_web::{HttpResponse, error};
+use actix_web_validator::{Error, PathConfig};
+
 use paperclip::actix::web;
 
 mod data_provider;
 mod resources;
 mod schemas;
 
+#[derive(serde::Serialize)]
+pub struct ValidationErrorJsonPayload {
+    pub message: String,
+    pub fields: Vec<String>,
+}
+
+/// Custom error handler
+impl From<&validator::ValidationErrors> for ValidationErrorJsonPayload {
```

Comment:
We already have error handler in main (line 100). I suggest not to touch anything here and add one more handler to main, near the previous one. And the implementation should be the same, it's also Validation error. This way, we can reuse it in all the modules

FYI
We do not use general approach here, we decided to go with rich errors as it's defined in Rosetta RPC
https://www.rosetta-api.org/docs/models/Error.html

Reply:
Made some changes, could you recheck please?